### PR TITLE
Switch to using sbt-circe-org

### PR DESCRIPTION
**Pull request by zarthross**
Starts to address  https://github.com/circe/circe-yaml/issues/344

cc: @armanbilge

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -21,42 +12,42 @@ val Versions = new {
...
-//  .settings(commonSettings)
+val root = tlCrossRootProject.aggregate(
+  `circe-yaml-common`,
+  `circe-yaml`,
+  `circe-yaml-v12`
+)
+
+lazy val `circe-yaml-common` = crossProject(JVMPlatform)
+  .crossType(CrossType.Pure)
+  .withoutSuffixFor(JVMPlatform)
```
Comment (armanbilge) [inline build.sbt:37]: Why make everything a `crossProject`? This project will never cross-compile for JS (assuming it sticks with the snake yaml parser).

Comment (zarthross) [reply build.sbt]: Following changes in circe-fs2.  Wasn't sure if it was needed.  Undone now.

File: build.sbt [hunk 2]
```diff
@@ -21,182 +13,70 @@ val Versions = new {
...
-        "io.circe" %% "circe-testing" % Versions.circe % Test,
-        "org.typelevel" %% "discipline-core" % Versions.discipline % Test,
-        "org.scalacheck" %% "scalacheck" % Versions.scalaCheck % Test,
-        "org.scalatest" %% "scalatest" % Versions.scalaTest % Test,
-        "org.scalatestplus" %% "scalacheck-1-15" % Versions.scalaTestPlus % Test
-      ),
-      mimaPreviousArtifacts := Versions.previousCirceYamls.map("io.circe" %% "circe-yaml" % _)
-    )
-    .enablePlugins(GhpagesPlugin)
+val docMappingsApiDir = settingKey[String]("Subdirectory in site target directory for API docs")
```
Comment (armanbilge) [inline build.sbt:24]: I think this can be removed.

**Review by zmccoy — APPROVED** [review]

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,13 +1,3 @@
...
-addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "1.1.1")
-addSbtPlugin("com.typesafe.sbt" % "sbt-site" % "1.4.1")
-addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.4.6")
-addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0")
-addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.9.3")
-addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "3.9.14")
-addSbtPlugin("com.codecommit" % "sbt-github-actions" % "0.13.0")
+addSbtPlugin("io.circe" % "sbt-circe-org" % "0.1.1")
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.5.0")
+addSbtPlugin("org.scoverage" % "sbt-scoverage" % "2.0.6")
```
Comment (armanbilge) [inline project/plugins.sbt:3]: These are provided by `sbt-circe-org`

**Review by armanbilge — APPROVED** [review]
LGTM, just a couple redundant things you can remove

**Comment by armanbilge** [comment]
Oh also: you may want to rename the default branch to `main` so that snapshotting works.

File: build.sbt (hunk 2 above)
Comment (zarthross) [reply build.sbt]: Done, nice catch

File: project/plugins.sbt (hunk 3 above)
Comment (zarthross) [reply project/plugins.sbt]: :+1:

**Comment by zarthross** [comment]
> Oh also: you may want to rename the default branch to `main` so that snapshotting works.

Done.  Thanks!

**Pull request merged**